actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -753,8 +753,8 @@ def apply_form_for_options!(object, options) # :nodoc:
       #   def labelled_form_with(**options, &block)
       #     form_with(**options.merge(builder: LabellingFormBuilder), &block)
       #   end
-      def form_with(model: false, scope: nil, url: nil, format: nil, **options, &block)
-        raise ArgumentError, "The :model argument cannot be nil" if model.nil?
+      def form_with(model: nil, scope: nil, url: nil, format: nil, **options, &block)
+        ActionView.deprecator.warn("Passing nil to the :model argument is deprecated and will be removed in Rails 7.3") if model.nil?
```

Comment:
```suggestion
        ActionView.deprecator.warn("Passing nil to the :model argument is deprecated and will raise in Rails 7.3") if model.nil?
```

Don't we still need the `false` default to differentiate when `nil` is passed and when `model` isn't specified?

Otherwise I think this will warn when doing `form_with(url: ...)`

Reply:
Ah right, that makes sense, thanks a bunch!